### PR TITLE
fix histograms for TIV variable

### PR DESCRIPTION
**Pull request by Marigold**
* Resolve https://github.com/LREN-CHUV/algorithm-repository/issues/10
* Add unit tests

```
"PARAM_query=SELECT \"tiv\",\"dataset\",\"gender\",\"agegroup\",\"alzheimerbroadcategory\" FROM cde_features_a WHERE \"tiv\" IS NOT NULL",
```
returns no results because all `tiv` values are NULL. This PR fixes that and returns error to `job_result` table.

### CODE REVIEW

File: python-histograms/tests/unit/test_histograms.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import mock
+import json
+from . import fixtures as fx
+from histograms import main, compute_categories, UserError
+
+
+@mock.patch('histograms.io_helper.fetch_data')
+@mock.patch('histograms.io_helper.get_results')
+@mock.patch('histograms.io_helper.save_results')
+def test_main(mock_save_results, mock_get_results, mock_fetch_data):
```
Comment (ludovicc) [inline python-histograms/tests/unit/test_histograms.py]: Yes, tests!

**Comment by Marigold** [comment]
With this change, call to `/mining/job` is returning
```
{
    "algorithm": "python-histograms",
    "error": "Cannot handle job results of shape ch.chuv.lren.woken.core.model.Shapes$highcharts$@15b89cc4 produced by function python-histograms with data Some(), error Some(Dependent variable tiv is empty.)",
    "jobId": "cf21e908-0f9a-41e6-9472-f71e7dbcfcb9",
    "node": "federation",
    "shape": "text/plain+error",
    "timestamp": 1521319557
}
```
is that acceptable?

**Comment by ludovicc** [comment]
No, you are returning an error message and mark it as a highchart json. It should have been a json that defines a highchart graph saying that there is no data for the variable. I will have a look and fix it

**Comment by Marigold** [comment]
Thanks for correcting me, I've changed the output type to `Shapes.ERROR` and woken now returns
```
{
    "algorithm": "python-histograms",
    "error": "Dependent variable tiv is empty.",
    "jobId": "270a4d45-7e69-4a48-8572-82a36cac87e2",
    "node": "federation",
    "shape": "text/plain+error",
    "timestamp": 1521368504
}
```
I hope it's correct now.

**Comment by Marigold** [comment]
Closing this in favor of https://github.com/LREN-CHUV/algorithm-repository/pull/13 with distributed mode.

**Pull request closed without merging**